--- /usr/local/cross/gcc/x86_64-pc-elf/include/c++/4.8.1/bits/stl_list.h	2013-07-23 22:14:04.736818225 -0400
+++ mod_stl_list.h	2013-07-26 14:53:49.154484992 -0400
@@ -53,16 +53,17 @@
  *  Do not attempt to use it directly. @headername{list}
  */
 
-#ifndef _STL_LIST_H
-#define _STL_LIST_H 1
+#ifndef _MOD_STL_LIST_H
+#define _MOD_STL_LIST_H 1
 
 #include <bits/concept_check.h>
 #if __cplusplus >= 201103L
 #include <initializer_list>
 #endif
 
-namespace std _GLIBCXX_VISIBILITY(default)
+namespace modstd _GLIBCXX_VISIBILITY(default)
 {
+#if 0
   namespace __detail
   {
   _GLIBCXX_BEGIN_NAMESPACE_VERSION
@@ -98,7 +99,7 @@
 
   _GLIBCXX_END_NAMESPACE_VERSION
   } // namespace detail
-
+#endif
 _GLIBCXX_BEGIN_NAMESPACE_CONTAINER
 
   /// An actual node in the %list.
@@ -330,8 +331,8 @@
       _List_impl _M_impl;
 
       _List_node<_Tp>*
-      _M_get_node()
-      { return _M_impl._Node_alloc_type::allocate(1); }
+      _M_get_node(const _Tp& __x)
+      { return _M_impl._Node_alloc_type::allocate(__x); }
 
       void
       _M_put_node(_List_node<_Tp>* __p)
@@ -378,7 +379,8 @@
       { _M_clear(); }
 
       void
-      _M_clear();
+      _M_clear()
+      { reinterpret_cast<std::_List_base<_Tp,_Alloc>*>(this)->_M_clear(); }
 
       void
       _M_init()
@@ -481,7 +483,7 @@
       _Node*
       _M_create_node(const value_type& __x)
       {
-	_Node* __p = this->_M_get_node();
+	_Node* __p = this->_M_get_node(__x);
 	__try
 	  {
 	    _M_get_Tp_allocator().construct
@@ -499,7 +501,7 @@
         _Node*
         _M_create_node(_Args&&... __args)
 	{
-	  _Node* __p = this->_M_get_node();
+	  _Node* __p = this->_M_get_node(__args...);
 	  __try
 	    {
 	      _M_get_Node_allocator().construct(__p,
