--- ../src/extern/linux-3.9.4/include/linux/cpumask.h	2013-07-10 19:53:28.594033402 -0400
+++ cpumask.h	2013-07-10 19:54:20.712077063 -0400
@@ -146,8 +146,8 @@
 	for ((cpu) = 0; (cpu) < 1; (cpu)++, (void)mask)
 #define for_each_cpu_not(cpu, mask)		\
 	for ((cpu) = 0; (cpu) < 1; (cpu)++, (void)mask)
-#define for_each_cpu_and(cpu, mask, and)	\
-	for ((cpu) = 0; (cpu) < 1; (cpu)++, (void)mask, (void)and)
+#define for_each_cpu_and(cpu, mask, _and)	\
+	for ((cpu) = 0; (cpu) < 1; (cpu)++, (void)mask, (void)_and)
 #else
 /**
  * cpumask_first - get the first cpu in a cpumask
@@ -231,9 +231,9 @@
  *
  * After the loop, cpu is >= nr_cpu_ids.
  */
-#define for_each_cpu_and(cpu, mask, and)				\
+#define for_each_cpu_and(cpu, mask, _and)				\
 	for ((cpu) = -1;						\
-		(cpu) = cpumask_next_and((cpu), (mask), (and)),		\
+		(cpu) = cpumask_next_and((cpu), (mask), (_and)),		\
 		(cpu) < nr_cpu_ids;)
 #endif /* SMP */
 
